app/src/components/header/header.jsx

```diff
@@ -2,18 +2,60 @@ import styles from "./header.module.css"
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
 import {faUser, faX} from '@fortawesome/free-solid-svg-icons'
 import Modal from "../modal/modal"
+import { useState } from "react"
+
+import Cookies from 'universal-cookie';
+
+// firebase utils
+import {newUser, existingUser} from "../../utils/firebase_utils"
 
 export default function Header(){
 
+    const [email, setEmail] = useState("")
+    const [username, setUsername] = useState("Login")
+    const [password, setPassword] = useState("")
+
+
+    //state for modal show
+    const [isOpen, setIsOpen] = useState(false)
+
+    // set up cookies
```

Comment:
@enoch-lee This is the solution I came up with for user auth. I use a universal cookie to store the user details, so that page refreshes still maintain auth state. Is this a good approach?